core-bundle/contao/dca/tl_user.php

```diff
@@ -351,7 +352,7 @@
 		(
 			'input_field_callback'    => array('tl_user', 'sessionField'),
 			'eval'                    => array('doNotShow'=>true, 'doNotCopy'=>true),
-			'sql'                     => "blob NULL"
+			'sql'                     => array('type' => 'blob', 'length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB, 'notnull' => false)
```

Comment:
Why not just this?

```suggestion
			'sql'                     => "mediumblob NULL"
```

Reply:
We should always use the array notation. We are already using the array notation in other places as well.

Reply:
Yes, for boolean fields that cannot be configured with simple strings. But certainly not for everything? `mediumblob NULL` is much shorter and easier to understand than `array('type' => 'blob', 'length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB, 'notnull' => false)`, don't you think?

Reply:
> Yes, for boolean fields that cannot be configured with simple strings.

Why can they not configured with simple strings? Instead of writing `['type' => 'boolean', 'default' => false]` you can also write `'tinyint DEFAULT 0'`.

> mediumblob NULL is much shorter and easier to understand than array('type' => 'blob', 'length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB, 'notnull' => false), don't you think?

The array notation is more descriptive and can be parsed more easily. Plus you can then rely on DBAL to create the correct notation for your platform.

Reply:
> Instead of writing `['type' => 'boolean', 'default' => false]` you can also write `'tinyint DEFAULT 0'`.

Yes, but then our parser (or maybe their parser?) does not handle the field correctly anymore, which is why we have to use the array notation for boolean fields.

Reply:
> Yes, but then our parser (or maybe their parser?) does not handle the field correctly anymore, which is why we have to use the array notation for boolean fields.

I cannot confirm that. Using `tinyint DEFAULT 0` works just as well.

In my original PR to introduce the boolean fields I used the array notation for the same reasons as I outlined here and considere it best practice.